client/app/cloud/project/compute/infrastructure/privateNetwork/cloud-project-compute-infrastructure-privateNetwork-list.directive.js

```diff
@@ -98,6 +107,32 @@ class PrivateNetworkListCtrl {
       .finally(() => { this.loaders.vrack.get = false; });
   }
 
+  linkProjectToVrack(selectedVrack) {
+    this.VrackService.linkCloudProjectToVrack(selectedVrack.serviceName, this.serviceName).$promise
+      .then(vrackTaskId => this.startVrackTaskPolling(this.models.vrack.id, vrackTaskId).$promise)
+      .then(() => {
+        this.CloudMessage.success(this.$translate.instant('cpci_private_network_add_vrack_success'));
+      })
+      .catch((err) => {
+        if (err === 'cancel') {
+          return;
+        }
+        this.CloudMessage.error(this.$translate.instant('cpci_private_network_add_vrack_error'));
```

Comment:
Why not test the other way round ? 
```suggestion
       if (err !== 'cancel') {
        this.CloudMessage.error(this.$translate.instant('cpci_private_network_add_vrack_error'));
       }
```